core/src/main/java/org/mskcc/oncotree/utils/TumorTypesUtil.java

```diff
@@ -88,15 +90,58 @@ public static Properties getProperties() {
         return filtered;
     }
 
-    public static InputStream getTumorTypeInputStream() {
-        Properties properties = getProperties();
-        InputStream inputStream = null;
+    public static InputStream getTumorTypeInputStream(Map<String, TumorType> tumorTypes) {
+        String tsvAsString = getTsvFromTumorTypes(tumorTypes);
         try {
-            inputStream = new FileInputStream(properties.getProperty("tumor_type_file_path"));
-        } catch (FileNotFoundException e) {
-            e.printStackTrace();
+            return new ByteArrayInputStream(tsvAsString.getBytes("UTF-8"));
+        } catch (UnsupportedEncodingException e) {
+           e.printStackTrace();
+        }
+        return null;
+    }
+
+    private static String getTsvFromTumorTypes(Map<String, TumorType> tumorTypes) {
+        List<String> rows = new ArrayList<>();
+        rows.add(TSV_HEADER);
+        // could hard code this instead of for loop
+        for (String code : tumorTypes.keySet()) {
+            TumorType tumorType = tumorTypes.get(code);
+            // skip the root node, "TISSUE". Just add it's children
+            Map<String, TumorType> children = tumorType.getChildren();
+            for (String childCode : children.keySet()) {
+                addTumorTypeToRows(children.get(childCode), rows, new ArrayList<String>());
+            }
+        }
+        return StringUtils.join(rows, "\n");
```

Comment:
This does not put a newline on the last row. Should each line be terminated by a new line? I generally do (to keep all lines in the same format), but not sure what the standard is.